dds_glossary/services.py

```diff
@@ -93,6 +96,61 @@ def __init__(
         self.data_dir.mkdir(parents=True, exist_ok=True)
         onto_path.append(str(self.data_dir))
 
+    @staticmethod
+    def check_concept_scheme_exists(
+        concept_scheme: ConceptScheme | None,
+        concept_scheme_iri: str,
+    ) -> ConceptScheme:
+        """
+        Check if the concept scheme exists, otherwise raise an HTTPException.
+
+        Args:
+            concept_scheme (ConceptScheme | None): The concept scheme to check.
+            concept_scheme_iri (str): The concept scheme IRI.
+
+        Returns:
+            ConceptScheme: The concept scheme, just to make mypy happy!
+
+        Raises:
+            HTTPException: If the concept scheme does not exist.
+        """
+        if concept_scheme is None:
+            raise HTTPException(
+                status_code=HTTPStatus.NOT_FOUND,
+                detail=f"Concept scheme {concept_scheme_iri} not found.",
+            )
+        return concept_scheme
+
+    @staticmethod
+    def get_scheme_collections(members: list[Member]) -> list[Member]:
+        """
+        Get the collections from the members.
+
+        Args:
+            members (list[Member]): The members.
+
+        Returns:
+            list[Member]: The collections.
+        """
+        return [
+            member for member in members if member.member_type == MemberType.COLLECTION
+        ]
+
+    @staticmethod
+    def get_scheme_concepts(members: list[Member]) -> list[Member]:
+        """
+        Get the concepts from the members.
+
+        Args:
+            members (list[Member]): The members.
+
+        Returns:
+            list[Member]: The concepts.
+        """
+        return [
+            member for member in members if member.member_type == MemberType.CONCEPT
+        ]
```

Comment:
Can we make this into one method? e.g.:

```suggestion
    @staticmethod
    def get_scheme_by_type(members: list[Member], member_type) -> list[Member]:
        """
        Filter the members list by member_type

        Args:
            members (list[Member]): The members.
            member_type (…): the member type

        Returns:
            list[Member]: The filtered list.
        """
        return [
            member for member in members if member.member_type == member_type
        ]
```

Reply:
Resolved in ae5da76153063e62a5b407f1768ceecff042c775.